packages/amesos2/test/solvers/KLU2_UnitTests.cpp

```diff
@@ -596,6 +608,138 @@ namespace {
     } // end if numProcs = 2
   }
 
+#if defined(HAVE_AMESOS2_EPETRA)
+  //! @test Test for Epetra non-contiguous GIDs.
+  TEUCHOS_UNIT_TEST_TEMPLATE_3_DECL( KLU2, NonContigGIDEpetra, SCALAR, LO, GO )
+  {
+    typedef Epetra_CrsMatrix MAT;
+    typedef ScalarTraits<SCALAR> ST;
+    typedef Epetra_MultiVector MV;
+    typedef typename ST::magnitudeType Mag;
+    typedef Epetra_Map map_type;
+
+    using Tpetra::global_size_t;
+    using Teuchos::tuple;
+    using Teuchos::RCP;
+    using Teuchos::rcp;
+    using Scalar = SCALAR;
+
+    RCP<const Comm<int> > comm = Tpetra::getDefaultComm();
+
+    size_t myRank = comm->getRank();
+    const global_size_t numProcs = comm->getSize();
+
+    // Unit test created for 2 processes
+    if ( numProcs == 2 ) {
+
+      const global_size_t numVectors = 1;
+      const global_size_t nrows = 6;
+
+      const int numGlobalEntries = nrows;
+      const LO numLocalEntries = nrows / numProcs;
+
+      // Create non-contiguous Map
+      // This example: np 2 leads to GIDS: proc0 - 0,2,4  proc 1 - 1,3,5
+      Teuchos::Array<int> elementList(numLocalEntries);
+      for ( LO k = 0; k < numLocalEntries; ++k ) {
+        elementList[k] = myRank + k*numProcs + 4*myRank;
+      }
+
+      RCP< const map_type > map = rcp( new Epetra_Map(numGlobalEntries, numLocalEntries, elementList.data(), 0, Epetra_MpiComm(MPI_COMM_WORLD) ));
```

Comment:
You may need to replace ``MPI_COMM_WORLD`` with something like ``to_epetra_comm(comm)``.

Reply:
I used `#ifdef HAVE_MPI` to distinguish between serial and parallel, should work.